packages/slate-editor/src/extensions/web-bookmark/transforms/convertWebBookmark.ts

```diff
@@ -13,6 +14,18 @@ export function convertWebBookmark(
     presentation: Presentation,
 ) {
     if (presentation === 'embed') {
+        if (element.oembed.type === 'video') {
+            Transforms.setNodes<VideoNode>(
+                editor,
+                createVideoBookmark({ oembed: element.oembed, url: element.url }),
+                {
+                    at: [],
+                    match: (node) => node === element,
+                },
+            );
+            return;
+        }
+
```

Comment:
It's better to branch for the construction of the replacement element. As the rest of `setNodes()` call is exactly the same.

```ts
const converted = element.oembed.type === 'video' 
    ? createVideoBookmark(...)
    : createEmbed(...)
```